requirements.txt

```diff
@@ -1,6 +1,9 @@
 sqlite3@3.17 -DCMAKE_POSITION_INDEPENDENT_CODE=On
 boost@1.79 -DCMAKE_POSITION_INDEPENDENT_CODE=On --build
 half,https://github.com/pfultz2/half/archive/1.12.0.tar.gz -X header -H sha256:0a08660b68abb176ebc2a0cdf8de46e3182a7f46c66443bb80dbfaaec98cf969 --build
-ROCmSoftwarePlatform/rocMLIR@rocm-5.4.0 -H sha256:3823f455ee392118c3281e27d45fa0e5381f3c4070eb4e06ba13bc6b34a90a60 -DBUILD_FAT_LIBROCKCOMPILER=1
+ROCmSoftwarePlatform/rocMLIR@11f2e26f193d398cc67f00d9b8e5a58504a14092  -DBUILD_FAT_LIBROCKCOMPILER=1
+Dobiasd/FunctionalPlus@v0.2.18-p0
+https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz
 nlohmann/json@v3.9.1 -DJSON_MultipleHeaders=ON -DJSON_BuildTests=Off
-ROCmSoftwarePlatform/composable_kernel@eef009d001b928db1bb377a105c93b75e0dccc7b -DGPU_TARGETS="gfx900;gfx906;gfx908;gfx90a;gfx1030"
+Dobiasd/frugally-deep@v0.15.19-p0
```

Comment:
@shurale-nkn I understand your concern about adding this dependency. The alternative is for us to implement a rather performant CPU version of our heuristic which is a considerable undertaking. Please suggest an acceptable alternative

Reply:
@JehandadKhan Using of this library can lead to unpredictable consequences. At least its code must be carefully analyzed for mistakes, vulnerabilities, license violations (I've seen a link to stackoverflow in the code), etc. I'm also against using things like `Dobiasd/FunctionalPlus` because we use c++, not python.

Reply:
Are we actually limited to CPU version? We are literally a neural net primitive library. Can't we use our own good stuff for that?

Reply:
> Please suggest an acceptable alternative

Maybe we can create and use a fork? That would work as a firewall from any unwanted changes at least. But we'll have to maintain that fork.

Reply:
> Are we actually limited to CPU version? We are literally a neural net primitive library. Can't we use our own good stuff for that?

Yes we would prefer not to allocate resources on the GPU

Reply:
Could we use workspace for that?